Mac OS X: ld: symbol(s) not found for architecture x86_64

Hi there!

Today's compilation attempt has failed:

Darwin blackice 15.6.0 Darwin Kernel Version 15.6.0: Sun Jun  4 21:43:07 PDT 2017; root:xnu-3248.70.3~1/RELEASE_X86_64 x86_64
...
Mac OS X 10.11.6 El Cap.
...
Latest commit e4144b3  2 days ago @flatcap flatcap resync translations
...
configure: Summary of build options:

  Version:           20170714
  Host OS:           darwin15.6.0
  Install prefix:    /opt
  Compiler:          gcc
  CFlags:            -Wall -pedantic -Wno-long-long -g -O2 -fno-delete-null-pointer-checks -I/opt/include -I/opt/local/include -I/opt/sdk/include -I/var/sdk/include -I/var/include -I/usr/include -I/usr/local/include -I/var/include -I/var -I/usr/include/gnutls -I/usr/lib/ssl/include -I/usr/include/ncursesw -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/var/stash/include -I.  -I/opt/local/include -I/opt/local/include/openssl/include
  LDFlags:           -L/opt/lib -L/opt/local/lib -L/opt/sdk/lib -L/var/sdk/lib -L/opt/lib -L/usr/local/lib -L/usr/lib -L/var/lib -L/lib -L. -L/opt/local/include/openssl/lib
  Libs:              -lidn -lsasl2  -llua  -lnotmuch -lncursesw -L/opt/local/lib -dynamic -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err -lssl -lcrypto -lz -liconv
  Header cache libs: -Lhcache -lhcache  -lkyotocabinet

  GPGME:             yes
  PGP:               yes
  SMIME:             yes
  Notmuch:           yes
  Header Cache(s):   kyotocabinet 
  Lua:               yes
...
Making install in hcache
gcc  -Wall -pedantic -Wno-long-long -g -O2 -fno-delete-null-pointer-checks  -L/opt/lib -L/opt/local/lib -L/opt/sdk/lib -L/var/sdk/lib -L/opt/lib -L/usr/local/lib -L/usr/lib -L/var/lib -L/lib -L. -L/opt/local/include/openssl/lib -o mutt_md5 mutt_md5.o -L../lib -lmutt -lidn -lsasl2  -llua  -lnotmuch -lncursesw -L/opt/local/lib -dynamic -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err -lssl -lcrypto -lz -liconv
ld: warning: directory not found for option '-L/opt/sdk/lib'
ld: warning: directory not found for option '-L/var/sdk/lib'
ld: warning: directory not found for option '-L/lib'
ld: warning: directory not found for option '-L/opt/local/include/openssl/lib'
Undefined symbols for architecture x86_64:
  "_libintl_gettext", referenced from:
      _default_perror in libmutt.a(message.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [mutt_md5] Error 1
make[1]: *** [install-recursive] Error 1
make: *** [install] Error 2

I confirm that I can reproduce on FreeBSD:

```
/usr/local/libexec/ccache/cc -DHAVE_CONFIG_H -I. -I..  -I..  -I/usr/local/include -I/usr/local/include  -Wall -pedantic -Wno-long-long -O2 -g -Wall -MT mutt_md5.o -MD -MP -MF .deps/mutt_md5.Tpo -c -o mutt_md5.o mutt_md5.c
mv -f .deps/mutt_md5.Tpo .deps/mutt_md5.Po
/usr/local/libexec/ccache/cc  -Wall -pedantic -Wno-long-long -O2 -g -Wall  -L/usr/local/lib -L/usr/local/lib -o mutt_md5 mutt_md5.o -L../lib -lmutt -lncursesw -lssl -lcrypto -lz
../lib/libmutt.a(message.o): In function `default_perror':
/usr/home/gahr/github/neomutt/lib/message.c:90: undefined reference to `libintl_gettext'
cc: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/hcache/mutt_md5.c
+++ b/hcache/mutt_md5.c
@@ -32,7 +32,7 @@
 #include <stddef.h>
 #include <stdbool.h>
 #include <string.h>
-#include "lib/lib.h"
+#include "lib/md5.h"
 
 /* local md5 equivalent for header cache versioning */
 int main(void)

